CNV-6749 New UI Flow for Virtual Machines

lmandavi opened the pull request:
This PR is associated with [CNV-6749](https://issues.redhat.com/browse/CNV-6749). In the 4.7 release, the UI flow for virtual machines has been redesigned. The following doc content has been revamped and updated to reflect the new UI flow. New doc content was added on using virtual machine templates in the web console.

The doc content updates include topics within each of these areas:
Creating virtual machines
Editing virtual machines
Deleting virtual machines

lmandavi commented:
@yaacov - this PR is ready for SME review:

Creating virtual machines:
https://deploy-preview-29092--osdocs.netlify.app/openshift-enterprise/latest/virt/virtual_machines/virt-create-vms.html

Editing virtual machines:
https://deploy-preview-29092--osdocs.netlify.app/openshift-enterprise/latest/virt/virtual_machines/virt-edit-vms.html

Deleting virtual machines:
https://deploy-preview-29092--osdocs.netlify.app/openshift-enterprise/latest/virt/virtual_machines/virt-delete-vms.html

yaacov commented on modules/virt-edit-cdrom-vm.adoc:
```diff
@@ -19,27 +19,21 @@ endif::[]
+. Click the *Disks* tab.
 
-* If CD-ROMs are unavailable for editing, the following message displays: *The virtual machine doesn't have any CD-ROMs attached.*
+. To edit a CD-ROM configuration, click the Options {kebab} menu for the CD-ROM and select *Edit CD-ROM.*
```
```suggestion
. To edit a CD-ROM configuration, click the Options {kebab} menu for the CD-ROM and select *Edit*
```

lmandavi replied:
@yaacov Edit was made. Appreciate the quick turnaround for your review :-)

yaacov commented on modules/virt-edit-cdrom-vm.adoc:
```diff
@@ -19,27 +19,21 @@ endif::[]
+. To edit a CD-ROM configuration, click the Options {kebab} menu for the CD-ROM and select *Edit CD-ROM.*
 
-* If there are CD-ROMs available, you can remove a CD-ROM by clicking *-*.
+. In the *Edit CD-ROM* window, edit the fields: *Source*, *Persistent Volume Claim*, *Name*, *Type*, and *Interface*.
```
```suggestion
. In the *Edit Disk* window, edit the fields: *Source*, *Persistent Volume Claim*, *Name*, *Type*, and *Interface*.
```

yaacov commented:
LGTM

lmandavi commented:
Thanks @yaacov!

lmandavi commented:
@gouyang This PR is ready for QE review. Please review and provide your input. Please note that for the **Add Network Interface** window, I noted today that we provide 3 drop-down options for users for Type: Masquerade, Bridge, SR-IOV. I am looking into updating the doc for that information as part of CNV-6750-usability-enhancements. Thanks!

gouyang commented:
```
When you create your virtual machine using the wizard, your virtual machine’s storage medium must support Read-Write-Many (RWX) PVCs.
```
This sentence is showing on preview doc but not in the PR, it's not correct, it depends on what's the storageClass is in use.

gouyang commented:
Others are LGTM

lmandavi commented:
@gouyang Requesting your review.

Item 1
I added information on the four methods for selecting a boot source. It helps the user to plan ahead before actually performing the tasks in the procedure:

https://deploy-preview-29092--osdocs.netlify.app/openshift-enterprise/latest/virt/vm_templates/virt-creating-vm-template.html#virt-adding-a-boot-source-web_virt-creating-vm-template

Item 2
I removed that prerequisite text on storage medium because:
1) It is information that was present in the CNV 4.5 docs and it is incomplete (mentions only about one type of permission)
2) We have detailed information on storage access modes and volume modes in our doc:https://docs.openshift.com/container-platform/4.6/virt/virtual_machines/virtual_disks/virt-storage-defaults-for-datavolumes.html
3) We mention the 3 types of permission in our doc

gouyang commented:
> @gouyang Requesting your review.
> 
> Item 1
> I added information on the four methods for selecting a boot source. It helps the user to plan ahead before actually performing the tasks in the procedure:
> 
> https://deploy-preview-29092--osdocs.netlify.app/openshift-enterprise/latest/virt/vm_templates/virt-creating-vm-template.html#virt-adding-a-boot-source-web_virt-creating-vm-template
> 

The four methods should be `Upload local file, Import via URL, Clone existing PVC, or Import via Registry`,  `PXE (network boot - adds network interface)` is not there.

> Item 2
> I removed that prerequisite text on storage medium because:
> 
>     1. It is information that was present in the CNV 4.5 docs and it is incomplete (mentions only about one type of permission)
> 
>     2. We have detailed information on storage access modes and volume modes in our doc:https://docs.openshift.com/container-platform/4.6/virt/virtual_machines/virtual_disks/virt-storage-defaults-for-datavolumes.html
> 
>     3. We mention the 3 types of permission in our doc

Looks good to me

gouyang commented:
LGTM

sjhala-ccs commented on modules/virt-editing-vm-web.adoc:
```diff
@@ -14,31 +14,40 @@
 = Editing a {object} in the web console
 
-Edit select values of a {object} that displays in the *{object-vm-overview}* screen of the web console by clicking on the pencil icon next to the relevant field. Other values can be edited using the CLI.
+Edit select values of a {object} that displays in the *{object-vm-details}* screen of the web console by clicking on the pencil icon next to the relevant field. Other values can be edited using the CLI.
```
Suggestion: "Edit select values of a {object} in the web console by clicking the pencil icon next to the relevant field."

lmandavi replied:
Change made.

sjhala-ccs commented on modules/virt-editing-vm-web.adoc:
```diff
@@ -14,31 +14,40 @@
-:object-gui: Virtual Machines
-:object-vm-overview: VM Template Details
+:object-gui: Templates
+:object-vm-overview: Virtual Machine Overview
```
Shouldn't the value of this variable be VM Template Details?

lmandavi replied:
I believe the correct values are displaying for each conditional expression to account for the difference between VMs and VM Templates:

ifeval::["{context}" == "virt-editing-vm-template"]
:virt-vm-template:
:object: virtual machine template
:object-gui: Templates
:object-vm-details: VM Template Details
endif::[]

ifeval::["{context}" == "virt-edit-vms"]
:virt-vm:
:object: virtual machine
:object-gui: Virtual Machines
:object-vm-details: Details
endif::[]

sjhala-ccs commented on modules/virt-editing-vm-web.adoc:
```diff
@@ -14,31 +14,40 @@
 :object-gui: Virtual Machines
 :object-vm-overview: Virtual Machine Overview
+:object-vm-details: Details
+:object-details: Details
```
This conditional is only defined for the vm context and not the template context. It is never used in the reusable procedure content.

lmandavi replied:
Yes, so here is the edited set of conditionals:

ifeval::["{context}" == "virt-editing-vm-template"]
:virt-vm-template:
:object: virtual machine template
:object-gui: Templates
:object-vm-details: VM Template Details
endif::[]

ifeval::["{context}" == "virt-edit-vms"]
:virt-vm:
:object: virtual machine
:object-gui: Virtual Machines
:object-vm-details: Details
endif::[]

sjhala-ccs commented on modules/virt-editing-vm-web.adoc:
```diff
@@ -14,31 +14,40 @@
-:object-vm-overview: VM Template Details
+:object-gui: Templates
+:object-vm-overview: Virtual Machine Overview
+:object-vm-details: VM Template Details
```
Is adding this new conditional variable necessary?

lmandavi replied:
This conditional is being used in the text:
= Editing a {object} in the web console

Edit select values of a {object} that displays in the *{object-vm-details}* screen of the web console by clicking on the pencil icon next to the relevant field. Other values can be edited using the CLI.

aburdenthehand replied:
@lmandavi  That step seems to have been rewritten and the `object-vm-detail` is no longer used in any of these modules. If that remains the case then it should be removed from the variable set/unset sections for each of these modules. 
However, I believe @sjhala-ccs is saying in comments further up that you could use the variable in those steps to improve that step in the procedure. In which case, please keep the setting/unsetting of that `object-vm-detail` variable in the relevant modules.

sjhala-ccs commented on modules/virt-editing-vm-yaml-web.adoc:
```diff
@@ -1,14 +1,24 @@
-Not all parameters can be updated. If you edit values that cannot be changed and click *Save*, an error message indicates the parameter that was not able to be updated.
+Using the web console, edit the YAML configuration of a virtual machine. Some parameters are not editable. If you edit values that cannot be changed and click *Save*, an error message indicates the parameter that was not able to be updated.
 
 The YAML configuration can be edited while the virtual machine is *Running*, however the changes will only take effect after the virtual machine has been stopped and started again.
```
Passive voice.

Suggestion: If the virtual machine is running, changes will not take effect until you restart the virtual machine.

lmandavi replied:
Removed the sentence with the passive voice. Incorporated your suggestion by combining two sentences into one:

If you edit the YAML configuration while the virtual machine is running, changes will not take effect until you restart the virtual machine.

sjhala-ccs commented on modules/virt-editing-vm-yaml-web.adoc:
```diff
@@ -1,14 +1,24 @@
-Using the web console, edit the YAML configuration of a virtual machine.
-
-Not all parameters can be updated. If you edit values that cannot be changed and click *Save*, an error message indicates the parameter that was not able to be updated.
+Using the web console, edit the YAML configuration of a virtual machine. Some parameters are not editable. If you edit values that cannot be changed and click *Save*, an error message indicates the parameter that was not able to be updated.
```
Suggestion: You can edit the YAML configuration of a virtual machine in the web console. Some parameters cannot be modified. If you click *Save* with an invalid configuration, an error message indicates the parameter that cannot be changed.

lmandavi replied:
Change made.

sjhala-ccs commented on modules/virt-editing-vm-yaml-web.adoc:
```diff
@@ -1,14 +1,24 @@
 // * virt/virtual_machines/virt-edit-vms.adoc
+//Establishing conditional expressions
+//
+ifeval::["{context}" == "virt-edit-vms"]
```
Maybe I am missing something but do you really need conditionals in this module? The content seems to be hard coded for VMs.

lmandavi replied:
The {object} conditional expression is used in the virt-editing-vm-yaml-web.adoc module, therefore retaining it. Removing the  unused conditional expressions.

lmandavi replied:
ifeval::["{context}" == "virt-edit-vms"]
:object: virtual machine
endif::[]

ifeval::["{context}" == "virt-editing-vm-template"]
:object: virtual machine template
endif::[]

[id="virt-editing-vm-yaml-web_{context}"]

sjhala-ccs commented on modules/virt-vm-wizard-fields-web.adoc:
```diff
@@ -89,6 +93,6 @@ endif::[]
 |Start this virtual machine after creation.
 |
-|Select to automatically start the virtual machine upon creation.
+|This checkbox is selected by default and the virtual machine starts running after creation. Clear the checkbox if you do not want the virtual machine to start when it is created.
```
Suggestion: Checkbox to indicate if the virtual machine should start after it is created. It is selected by default.

lmandavi replied:
In referring to the IBM Style Guide, they recommend "clear the checkbox" - which is the usage in this case. Here I am describing the default behavior first followed by how to undo the default behavior. Prefer this approach but I appreciate your suggestion.first

sjhala-ccs commented on virt/virtual_machines/virt-delete-vms.adoc:
```diff
@@ -4,8 +4,7 @@ include::modules/virt-document-attributes.adoc[]
 
-You can delete a virtual machine from the web console or by using the `oc`
-command-line interface.
+You can delete a virtual machine from the web console or by using the `oc` command-line interface.
```
s/command-line/command line

lmandavi replied:
Change made.

aburdenthehand replied:
Still present in most recent commit

lmandavi replied:
Removed the hyphen.

lmandavi replied:
Victim of the GitHub outage. Change had been made but not caught by GitHub after the push.

lmandavi replied:
From the .adoc file:
[id="virt-delete-vms"]
= Deleting virtual machines
include::modules/virt-document-attributes.adoc[]
:context: virt-delete-vms
toc::[]

You can delete a virtual machine from the web console or by using the `oc` command line interface.

sjhala-ccs commented on virt/virtual_machines/virt-edit-vms.adoc:
```diff
@@ -4,9 +4,7 @@ include::modules/virt-document-attributes.adoc[]
-You can update a virtual machine configuration using either the YAML editor in the
-web console or the OpenShift client on the command line. You can also update
-a subset of the parameters in the *Virtual Machine Overview* of the web console.
+You can update a virtual machine configuration using either the YAML editor in the web console or the OpenShift client on the command line. You can also update a subset of the parameters in the *Virtual Machine Details* screen that displays in the web console.
```
s/OpenShift client on the command line/`oc` command line interface

Suggestion to remove "that displays in the web console" in the second sentence.

lmandavi replied:
Removed.

sjhala-ccs commented on virt/virtual_machines/virt-create-vms.adoc:
```diff
@@ -7,28 +7,32 @@ toc::[]
+Instead, create a new namespace or use an existing namespace without the `openshift` prefix.
 ====
 
+When you create virtual machines from the web console, select a virtual machine template that is configured with a boot source. Virtual machine templates with a boot source are labeled as *Available boot source* or they display a customized label text. Using templates with an available boot source expedites the process for creating virtual machines.
```
s/process for creating/process of creating

lmandavi replied:
Change made.

sjhala-ccs reviewed (commented):
Handful of suggestions for you to consider. Looks good, otherwise. 

@aburdenthehand Can you please take a second pass at reviewing this PR, especially the conditional content? Thanks!

lmandavi commented on modules/virt-creating-vm-quick-start-web.adoc:
```diff
@@ -0,0 +1,29 @@
+[id="virt-creating-vm-quick-start-web_{context}"]
+= Using a Quick Start to create a virtual machine
+
+The web console provides Quick Starts with instructional guided tours for creating virtual machines. You can access Quick Starts by selecting the Help menu in the *Administrator* perspective to view the Quick Starts catalog. When you click on a Quick Start tile and begin the tour, instructions display in the Quick Start panel and guide you to complete the tasks.
```
Modified per suggestion.

lmandavi commented on modules/virt-adding-a-boot-source-web.adoc:
```diff
@@ -5,7 +5,23 @@
+
+* To import via URL, access to the web server with the operating system image is required. For example: the Red Hat Enterprise Linux web page with images.
+
+* To clone an existing PVC, access to the project with a PVC is is required.
```
Removed.

lmandavi commented on modules/virt-add-disk-to-vm.adoc:
```diff
@@ -17,7 +17,6 @@ ifeval::["{context}" == "virt-editing-vm-template"]
 :virt-vm-template:
 :object: virtual machine template
 :object-gui: Templates
-:object-vm-overview: VM Template Details
```
Step 3 in the procedure (virt-add-disk-to-vm.adoc):

Add a {object}. Please see below. Was the truncation in a different module perhaps...

Adding a virtual disk to a {object}
Use this procedure to add a virtual disk to a {object}.

Procedure
Click Workloads → Virtualization from the side menu.

Click the {object-gui} tab.

Select a {object}.

Click the Disks tab.

sjhala-ccs replied:
@lmandavi I meant we usually write this step as Select a {object} to open the {object-vm-overview} screen. It just sounds more complete to me that way. Any specific reason why you decided not to use that `object-vm-overview` conditional?

lmandavi replied:
The famous {object-overview} has been added.

lmandavi commented on modules/virt-add-nic-to-vm.adoc:
```diff
@@ -6,15 +6,16 @@
 :object: virtual machine template
 :object-gui: Templates
-:object-vm-overview: VM Template Details
+:object-vm-details: VM Template Details
```
Removed.

lmandavi replied:
Victim of the GitHub outage. Repushing the change again.

lmandavi replied:
From the .adoc file:
ifeval::["{context}" == "virt-edit-vms"]
:virt-vm:
:object: virtual machine
:object-gui: Virtual Machines
:object-vm-overview: Virtual Machine Overview
endif::[]

ifeval::["{context}" == "virt-editing-vm-template"]
:virt-vm-template:
:object: virtual machine template
:object-gui: Templates
:object-vm-overview: VM Template Details
endif::[]

[id="virt-vm-add-nic_{context}"]

= Adding a network interface to a {object}

Use this procedure to add a network interface to a {object}.

.Procedure

. Click *Workloads* -> *Virtualization* from the side menu.

. Click the *{object-gui}* tab.

. Select a {object} to open the *{object-vm-overview}* screen.

. Click the *Network Interfaces* tab.

lmandavi commented on modules/virt-add-disk-to-vm.adoc:
```diff
@@ -59,5 +63,5 @@ ifeval::["{context}" == "virt-editing-vm-template"]
 :object!:
 :object-gui!:
-:object-vm-overview!:
+:object-vm-details!:
```
Removed.

aburdenthehand replied:
Still there after most recent commit

lmandavi replied:
That is so odd. My local virt-add-disk-to-vm.adoc file shows the following

ifeval::["{context}" == "virt-edit-vms"]
:virt-vm:
:object: virtual machine
:object-gui: Virtual Machines
endif::[]

ifeval::["{context}" == "virt-editing-vm-template"]
:virt-vm-template:
:object: virtual machine template
:object-gui: Templates
endif::[]

[id="virt-vm-add-disk_{context}"]l virt-add-disk-to-vm.adoc file shows the following;

sjhala-ccs replied:
@lmandavi The `object-vm-details` variable is still showing up

lmandavi replied:
This particular item, I suspect, became a victim of the GitHub outage. The .adoc file that I am viewing does not have {object-vm-details} in it (as shown above). Repushing again and hopefully GitHub will grab it this time.

lmandavi replied:
From the .adoc file:
ifeval::["{context}" == "virt-edit-vms"]
:virt-vm:
:object: virtual machine
:object-gui: Virtual Machines
:object-vm-overview: Virtual Machine Overview
endif::[]

ifeval::["{context}" == "virt-editing-vm-template"]
:virt-vm-template:
:object: virtual machine template
:object-gui: Templates
:object-vm-overview: VM Template Details
endif::[]

[id="virt-vm-add-disk_{context}"]

= Adding a virtual disk to a {object}

Use this procedure to add a virtual disk to a {object}.

.Procedure

. Click *Workloads* -> *Virtualization* from the side menu.

. Click the *{object-gui}* tab.

. Select a {object} to open the *{object-vm-overview}* screen.

. Click the *Disks* tab.

lmandavi commented on modules/virt-creating-vm-quick-start-web.adoc:
```diff
@@ -0,0 +1,29 @@
+
+The web console provides Quick Starts with instructional guided tours for creating virtual machines. You can access Quick Starts by selecting the Help menu in the *Administrator* perspective to view the Quick Starts catalog. When you click on a Quick Start tile and begin the tour, instructions display in the Quick Start panel and guide you to complete the tasks.
+
+Tasks in a Quick Start begin with selecting a Red Hat template, adding a boot source, importing the operating system image, and saving the custom template. Then, you can select the custom template and create a virtual machine.
```
Reordered the sentences to reflect the task sequence summary.

lmandavi commented on modules/virt-creating-vm-quick-start-web.adoc:
```diff
@@ -0,0 +1,29 @@
+
+.Procedure
+
+. In the web console, click the Help menu.
```
Reordered to create one step.

lmandavi commented on modules/virt-creating-vm-quick-start-web.adoc:
```diff
@@ -0,0 +1,29 @@
+
+. Click on a tile in the Quick Starts catalog. For example: *Creating a Red Hat Linux Enterprise Linux virtual machine*.
+
+. Follow the instructions in the guided tour and complete the tasks for importing an operating system image and creating a virtual machine. The new virtual machine displays in the *Virtual Machines* tab.
```
Recasted the sentence to adhere to ISG.

lmandavi commented on modules/virt-creating-vm-wizard-web.adoc:
```diff
@@ -5,32 +5,50 @@
 = Running the virtual machine wizard to create a virtual machine
 
-The web console features a wizard that guides you through *General*, *Networking*, *Storage*, *Advanced*, and *Review* steps to simplify the process of creating virtual machines. All required fields are marked by a ++*++. When the required fields are completed, you can review and create your virtual machine.
+The web console features a wizard that guides you to select a virtual machine template and create a virtual machine. Red Hat virtual machine templates are preconfigured with an operating system image, default settings for the operating system, flavor (CPU and memory), and workload type (server). When templates are configured with a boot source, they are labeled with a customized label text or the ...
```
Incorporated the suggestion.

lmandavi commented on modules/virt-creating-vm-wizard-web.adoc:
```diff
@@ -5,32 +5,50 @@
-The web console features a wizard that guides you through *General*, *Networking*, *Storage*, *Advanced*, and *Review* steps to simplify the process of creating virtual machines. All required fields are marked by a ++*++. When the required fields are completed, you can review and create your virtual machine.
+The web console features a wizard that guides you to select a virtual machine template and create a virtual machine. Red Hat virtual machine templates are preconfigured with an operating system image, default settings for the operating system, flavor (CPU and memory), and workload type (server). When templates are configured with a boot source, they are labeled with a customized label text or the ...
+
+When you run the *Create virtual machine from template* wizard, select a template from a list of preconfigured templates, review the settings, and create the virtual machine. If you choose to customize your virtual machine, the wizard guides you through the *General*, *Networking*, *Storage*, *Advanced*, and *Review* steps. All required fields displayed by the wizard are marked by a ++*++.
```
Recasted the sentence.

lmandavi commented on modules/virt-creating-vm-wizard-web.adoc:
```diff
@@ -5,32 +5,50 @@
+. From the *Virtual Machines* tab or the *Templates* tab, click *Create* and select *Virtual Machine with Wizard*.
 
-.Prerequisites
+. In the *Select a template* step, select a template that is configured with a boot source.
```
Changed it to;
Select a template that is configured with a boot source.

lmandavi commented on modules/virt-creating-vm-wizard-web.adoc:
```diff
@@ -5,32 +5,50 @@
+. In the *Select a template* step, select a template that is configured with a boot source.
 
-* When you create your virtual machine using the wizard, your virtual machine's storage medium must support Read-Write-Many (RWX) PVCs.
+. Click *Next* to go to the *Review and create* step.
```
It is an actual step where the user reviews the settings and clicks "Create virtual machine"

sjhala-ccs replied:
@lmandavi I meant to ask if it was a "step" in the same way as "General", "Networking", etc. Is *Review and create* an actual UI element? Would rewording it as "Click *Next* to review your settings and create the virtual machine" make sense?

lmandavi replied:
HI @sjhala-ccs - Review is a UI element - it is listed in the left pane in a numbered list: 1) General 2) Networking 3) Storage 4) Advanced 5) Review.

Review and create, on the other hand, is the final step where the user creates the VM and exits the wizard.

In the wizard (speaking from memory because I don't have access to a server today), the title for the final step where the user clicks Create Virtual Machine button is:

Review and Create. 

So Review and Create is the final step after General, Networking, Storage etc  because it shows up as the final step.

lmandavi commented on modules/virt-creating-vm-wizard-web.adoc:
```diff
@@ -5,32 +5,50 @@
+. Click *Next* to go to the *Review and create* step.
 
-.Procedure
+. Clear the checkbox for *Start this virtual machine after creation* if the virtual machine should not be started now.
```
suggestion incorporated.

lmandavi commented on modules/virt-creating-vm-wizard-web.adoc:
```diff
@@ -5,32 +5,50 @@
-.Procedure
+. Clear the checkbox for *Start this virtual machine after creation* if the virtual machine should not be started now.
+
+. Click *Create virtual machine* and exit the wizard or continue with the wizard to customize the virtual machine.
```
Splitting the procedure is possible but it might require an additional module and some planning. Maybe consider this for the next release?

lmandavi commented on modules/virt-edit-cdrom-vm.adoc:
```diff
@@ -19,27 +19,21 @@ endif::[]
 
-. Select the *Overview* tab.
+. Click the *{object-gui}* tab.
 
```
@sjhala-ccs , @aburdenthehand Hi _ I don't have access today to a server build :-(  

The VM Overview screen belongs only the flow for VMs. 
Can a conditional expression be created such that a step shows up in Procedure A (for VMs) and not for Procedure B (for VM templates). If that is possible, then the additional step can be added such that it shows up only in the procedure for the VMs. Please advise.

sjhala-ccs replied:
@lmandavi Like I said in my original comment, you can add the "VM Overview screen" part of the sentence if you declare another conditional. For example, set `object-overview` to `Virtual Machine Overview` for the VM context and the same conditional to `Template Details` for the template context. Does that make sense?

lmandavi replied:
Let me know if I am missing out on something here, but here's what I am noting.

When you click on Virtual Machines, the VM Overview screen displays (step 1)

Click on Details, and the VM Details screen displays (this is step 2)

(So that's 2 steps in the flow to get to the VM Details).

When you click on Templates, the Template Details screen displays.

(So that's 1 step in the flow to get to the Template Details screen and in this flow we don't have the Overview step for Templates the way we have it for the VM flow).

In this scenario, how does having a conditional expression help when the flow for VM requires an additional step (from VM Overview, you click on VM Details) but the flow for VM Templates does not require that additional step?

Do let me know if I am missing something here!

sjhala-ccs replied:
@lmandavi I don't have access to a test env but iirc, you do not need to go to the VM Details screen to access the Disks tab (which is the next step). The different tabs are accessible from the VM Overview screen itself.

lmandavi replied:
The famous {object-overview} conditional is now present in three .adoc files: add a nic, add a disk, and edit cd-rom.

lmandavi commented on modules/virt-edit-cdrom-vm.adoc:
```diff
@@ -19,27 +19,21 @@ endif::[]
+. Click the *Disks* tab.
 
-* If CD-ROMs are unavailable for editing, the following message displays: *The virtual machine doesn't have any CD-ROMs attached.*
+. To edit a CD-ROM configuration, click the Options {kebab} menu for the CD-ROM and select *Edit.*
```
Incorporated the input.

lmandavi commented on modules/virt-editing-template-yaml-web.adoc:
```diff
@@ -5,25 +5,25 @@
 
-A confirmation message, which includes the updated version number for the object,
-shows the modification has been successful.
+A confirmation message, which includes the updated version number for the object, shows the modification has been successful.
```
Change made.

sjhala-ccs commented on modules/virt-edit-cdrom-vm.adoc:
```diff
@@ -9,37 +9,33 @@
+. Click *Workloads* -> *Virtualization* from the side menu.
 
-. Select the *Overview* tab.
+. Click the *{object-gui}* tab to open the *{object-overview}* screen.
```
The steps should be:
`. Click the *{object-gui}* tab`. 
`. Select a {object} to open the *{object-overview}* screen.`
`. Click the *Disks* tab.`

Attaching rendered output for VMs and Templates for reference.

![image](https://user-images.githubusercontent.com/64486364/107578270-88a05080-6bc1-11eb-8a04-9fd36bc1da3e.png)

![image](https://user-images.githubusercontent.com/64486364/107578393-ae2d5a00-6bc1-11eb-95e1-033165a897c4.png)

lmandavi replied:
This one definitely a victim of the GitHub outage! The changes never rippled through to show up in GitHub...

lmandavi replied:
Verified correct rendering:
https://deploy-preview-29092--osdocs.netlify.app/openshift-enterprise/latest/virt/vm_templates/virt-editing-vm-template.html#virt-vm-edit-cdrom_virt-editing-vm-template

https://deploy-preview-29092--osdocs.netlify.app/openshift-enterprise/latest/virt/virtual_machines/virt-edit-vms.html#virt-vm-edit-cdrom_virt-edit-vms

sjhala-ccs commented on modules/virt-edit-cdrom-vm.adoc:
```diff
@@ -9,37 +9,33 @@
+. Click *Workloads* -> *Virtualization* from the side menu.
 
-. Select the *Overview* tab.
+. Select a *{object-gui}* tab to open the *{object-overview}* screen.
```
Re-posting my earlier comment about step 2 

> The steps should be:
> `. Click the *{object-gui}* tab`.
> `. Select a {object} to open the *{object-overview}* screen.`

lmandavi replied:
Content updated.

lmandavi replied:
From the .adoc file:

[id="virt-vm-edit-cdrom_{context}"]

= Editing CD-ROMs for {object-gui}
Use the following procedure to edit CD-ROMs for {object}s.

.Procedure

. Click *Workloads* -> *Virtualization* from the side menu.

. Click the {object-gui} tab.

. Select a *{object}* to open the *{object-overview}* screen.

lmandavi commented on modules/virt-delete-vm-web.adoc:
```diff
@@ -15,7 +15,11 @@ When you delete a virtual machine, the data volume it uses is automatically dele
+
 . Click the *Virtual Machines* tab.
+
 . Click the &#8942; button of the virtual machine that you want to delete and select *Delete Virtual Machine*.
```
Content updated.

sjhala-ccs commented on modules/virt-edit-cdrom-vm.adoc:
```diff
@@ -9,37 +9,35 @@
+. Click *Workloads* -> *Virtualization* from the side menu.
 
-. Select the *Overview* tab.
+. Click the {object-gui} tab.
```
Bold formatting  - s/{object-gui}/*{object-gui}*

lmandavi replied:
Bold formatting applied.

sjhala-ccs commented on modules/virt-edit-cdrom-vm.adoc:
```diff
@@ -9,37 +9,35 @@
+. Click the {object-gui} tab.
 
-. To add or edit a CD-ROM configuration, click the pencil icon to the right of the *CD-ROMs* label. The *Edit CD-ROM* window opens.
+. Select a *{object}* to open the *{object-overview}* screen.
```
Remove bold formatting - s/*{object}*/{object}

lmandavi replied:
Bold formatting removed.

sjhala-ccs reviewed (commented):
One nit about bold text formatting

sjhala-ccs commented:
LGTM!

aburdenthehand commented on modules/virt-editing-vm-web.adoc:
```diff
@@ -14,31 +14,37 @@
+
+. Select a {object}.
+
+. Click the *object-vm-details* tab.
```
This needs to be wrapped in {} so it's recognise as a variable

sjhala-ccs replied:
Good catch!

lmandavi replied:
Super catch! I am on it.

codyhoag reviewed (approved):
CNV doc with pre-approval to merge. LGTM

codyhoag commented:
/cherrypick enterprise-4.7
